src/components/App.tsx

```diff
@@ -2741,6 +2743,86 @@ class App extends React.Component<AppProps, AppState> {
     }
   };
 
+  private handleEraser = (
+    event: PointerEvent,
+    pointerDownState: PointerDownState,
+    scenePointer: { x: number; y: number },
+  ) => {
+    const idsToUpdate: Array<string> = [];
+
+    const distance = distance2d(
+      pointerDownState.lastCoords.x,
+      pointerDownState.lastCoords.y,
+      scenePointer.x,
+      scenePointer.y,
+    );
+    const threshold = 10 / this.state.zoom.value;
+    const point = pointerDownState.lastCoords;
+    let samplingInterval = 0;
+    while (samplingInterval < distance) {
+      const hitElement = this.getElementAtPosition(point.x, point.y);
+      if (hitElement) {
+        idsToUpdate.push(hitElement.id);
+        if (event.altKey) {
+          if (pointerDownState.elementIdsToErase[hitElement.id]) {
+            pointerDownState.elementIdsToErase[hitElement.id] = false;
+          }
+        } else {
+          pointerDownState.elementIdsToErase[hitElement.id] = true;
+        }
+      }
+
+      // Calculate next point in the line at a distance of sampling interval
+      samplingInterval += threshold;
+
+      const distanceRatio = samplingInterval / distance;
+      const nextX =
+        (1 - distanceRatio) * point.x + distanceRatio * scenePointer.x;
+      const nextY =
+        (1 - distanceRatio) * point.y + distanceRatio * scenePointer.y;
+      point.x = nextX;
+      point.y = nextY;
```

Comment:
I'd not mutate the event because you never know where it'll be used after this function calls (potentially even later in this function). Let's just keep the positions as local variables